### DISCUSSION TITLE
Autosync?

### DISCUSSION THREAD

**Discussion by look-a-furry** [abraunegg/onedrive#2391] [Q&A] [answered]
I just have a simple question, are files synced automatically? or do I need to run --resync every time I need to upload a file to one drive? 
my SyncDir is an external hard drive with lots of files, disk images, backups, etc. 
Do I have to run --resync every time A file is modified or a backup is made? I have over a TB of files and I'd like them to sync automatically.

**Comment by abraunegg** [comment, marked as answer]
@ranfur8 
> I just have a simple question, are files synced automatically?

It all depends.

Are you running the application manually (`--synchronize`) all the time - or are you using a systemd service using `--monitor`?

If `--synchronize` - then files are *not* automatically synced. They will only be synced when you run the application.

If using a systemd service, they will be synced as per your configuration.

In both instances - please read the documentation as this is clearly articulated: https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md

> or do I need to run --resync every time I need to upload a file to one drive?

**As per the documentation - do not use `--resync` unless the application tells you to.**

https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#performing-a---resync
